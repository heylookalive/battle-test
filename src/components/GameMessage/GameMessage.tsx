import React from 'react';
import styled from 'styled-components';

type GameMessageProps = {
  playerWin: boolean;
  monsterWin: boolean;
  children: React.ReactNode;
};

const Message = styled.h2.attrs((props: GameMessageProps) => ({
  playerWin: props.playerWin,
  monsterWin: props.monsterWin,
}))`
  ${(props) => (props.playerWin ? `color: limegreen;` : null)}
  ${(props) => (props.monsterWin ? `color: red;` : null)}
`;

export default function GameMessage({
  playerWin,
  monsterWin,
  children,
}: GameMessageProps) {
  return (
    <Message
      playerWin={playerWin}
      monsterWin={monsterWin}
      data-testid="game-message"
    >
      {children}
    </Message>
  );
}
