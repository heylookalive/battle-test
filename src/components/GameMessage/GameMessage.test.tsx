import { render } from '@testing-library/react';
import React from 'react';
import GameMessage from './GameMessage';

describe('<GameMessage />', () => {
  it('renders as expected', () => {
    const { container } = render(
      <GameMessage playerWin={false} monsterWin={false}>
        Text
      </GameMessage>
    );
    expect(container).toMatchSnapshot();
  });
});
