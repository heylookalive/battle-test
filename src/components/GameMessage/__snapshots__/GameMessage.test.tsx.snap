// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameMessage /> renders as expected 1`] = `
<div>
  <h2
    class="sc-AxjAm fiqGYr"
    data-testid="game-message"
  >
    Text
  </h2>
</div>
`;
